crates/fsrt/src/forge_project.rs

```diff
@@ -0,0 +1,165 @@
+use crate::{collect_sourcefiles, ResolvedEntryPoint};
+use forge_analyzer::ctx::AppCtx;
+use forge_analyzer::definitions::{run_resolver, Environment, PackageData};
+use forge_loader::manifest::{Entrypoint, ForgeManifest, Resolved};
+use std::collections::{HashMap, HashSet};
+use std::path::{Path, PathBuf};
+use std::sync::Arc;
+use swc_core::common::sync::Lrc;
+use swc_core::common::{FileName, Globals, Mark, SourceFile, SourceMap, GLOBALS};
+use swc_core::ecma::ast::EsVersion;
+use swc_core::ecma::parser::{parse_file_as_module, Syntax, TsConfig};
+use swc_core::ecma::transforms::base::resolver;
+use swc_core::ecma::visit::FoldWith;
+use tracing::debug;
+
+pub(crate) trait ForgeProjectTrait<'a> {
+    fn load_file(&self, path: PathBuf, _: Arc<SourceMap>) -> Arc<SourceFile>;
```

Comment:
```suggestion
    fn load_file(&self, path: impl AsRef<Path>, _: Arc<SourceMap>) -> Arc<SourceFile>;
```

This way, we can pass `&str` and borrowed paths as the file name. Also, we are always borrowing `path`, so there's no need to have the caller create an owned version of the `Path`.

Reply:
fixed! thank you :)